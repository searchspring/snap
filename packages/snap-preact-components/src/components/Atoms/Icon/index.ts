export * from './Icon';
export * from './paths';
