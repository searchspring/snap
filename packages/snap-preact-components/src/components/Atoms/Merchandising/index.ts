export * from './Banner';
export * from './InlineBanner';
