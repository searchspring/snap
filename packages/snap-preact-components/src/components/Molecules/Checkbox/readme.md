## Checkbox

Renders an span and an `<Icon />` to create a customizable checkbox. checked, color, disabled, onClick, size and Icon (defaults to 'check-thin') props available. 

## Additional Info

Using the native prop, you can bypass the span and Icon component and just render a native HTML input element with the type checkbox. 

The Checked state can be handled both internally or externally, by using either the checked prop, or the startChecked prop. 
 
## Components Used
- Icon

## Usage

Default
```jsx
    <Checkbox checked={true} disabled={true} />
```

Native
```jsx
    <Checkbox native={true} />
```

