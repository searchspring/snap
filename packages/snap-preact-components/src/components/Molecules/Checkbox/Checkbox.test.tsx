import { h } from 'preact';

import { render } from '@testing-library/preact';
import userEvent from '@testing-library/user-event';

import { Checkbox } from './Checkbox';
import { iconPaths } from '../../Atoms/Icon';
import { ThemeProvider } from '../../../providers';

const globalTheme = {
	components: {
		checkbox: {
			className: 'global-theme-class',
		},
		icon: {
			color: 'blue',
		},
	},
};

const propTheme = {
	components: {
		checkbox: {
			className: 'props-theme-class',
		},
		icon: {
			color: 'red',
		},
	},
};

describe('Checkbox Component', () => {
	describe('styled', () => {
		it('renders', () => {
			const rendered = render(<Checkbox />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');

			expect(checkboxElement).toBeInTheDocument();
			expect(checkboxElement.classList.length).toBe(2);
			expect(checkboxElement.className.match(/disabled/)).toBeFalsy();
		});

		it('is not checked by default', () => {
			const rendered = render(<Checkbox />);
			const svgElement = rendered.container.querySelector('svg');

			expect(svgElement).not.toBeInTheDocument();
		});

		it('renders an icon when checked', () => {
			const rendered = render(<Checkbox checked />);
			const svgElement = rendered.container.querySelector('svg');

			expect(svgElement).toBeInTheDocument();
		});

		it('fires onClick prop when clicked', () => {
			const clickFn = jest.fn();

			const rendered = render(<Checkbox onClick={clickFn} />);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');

			userEvent.click(checkboxElement);
			expect(clickFn).toHaveBeenCalled();
		});

		it('renders with additional style using prop', () => {
			const style = {
				padding: '20px',
			};

			const rendered = render(<Checkbox style={style} />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			const styles = getComputedStyle(checkboxElement);

			expect(styles.padding).toBe(style.padding);
		});

		it('respects the disabled prop', () => {
			const clickFn = jest.fn();
			const rendered = render(<Checkbox disabled onClick={clickFn} />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');

			expect(checkboxElement.className.match(/disabled/)).toBeTruthy();

			userEvent.click(checkboxElement);
			expect(clickFn).not.toHaveBeenCalled();
		});

		it('passes props to icon', () => {
			const icon = 'close-thin';
			const color = 'blue';
			const size = '12px';

			const rendered = render(<Checkbox checked size={size} icon={icon} color={color} />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			const svg = checkboxElement.querySelector('svg');
			const path = svg.querySelector('path');

			const styles = getComputedStyle(svg);

			expect(styles.width).toBe(`calc(${size} - 30%)`);
			expect(styles.height).toBe(`calc(${size} - 30%)`);
			expect(styles.fill).toBe(color);
			expect(path).toHaveAttribute('d', iconPaths[icon]);
		});

		it('can disableStyles', () => {
			const rendered = render(<Checkbox disableStyles />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');

			expect(rendered.container).toBeInTheDocument();
			expect(checkboxElement.classList.length).toBe(1);
			expect(checkboxElement.className.match(/disabled/)).toBeFalsy();
		});

		it('is themeable with ThemeProvider', () => {
			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Checkbox checked />
				</ThemeProvider>
			);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			const iconElement = checkboxElement.querySelector('.ss__icon');
			const iconStyles = getComputedStyle(iconElement);

			expect(checkboxElement).toHaveClass(globalTheme.components.checkbox.className);
			expect(iconStyles.fill).toBe(globalTheme.components.icon.color);
		});

		it('is themeable with theme prop', () => {
			const rendered = render(<Checkbox checked theme={propTheme} />);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			const iconElement = checkboxElement.querySelector('.ss__icon');
			const iconStyles = getComputedStyle(iconElement);

			expect(checkboxElement).toHaveClass(propTheme.components.checkbox.className);
			expect(iconStyles.fill).toBe(propTheme.components.icon.color);
		});

		it('is themeable and theme prop overrides ThemeProvider', () => {
			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Checkbox checked theme={propTheme} />
				</ThemeProvider>
			);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			const iconElement = checkboxElement.querySelector('.ss__icon');
			const iconStyles = getComputedStyle(iconElement);

			expect(checkboxElement).toHaveClass(propTheme.components.checkbox.className);
			expect(checkboxElement).not.toHaveClass(globalTheme.components.checkbox.className);
			expect(iconStyles.fill).toBe(propTheme.components.icon.color);
		});
	});

	describe('native', () => {
		it('renders', () => {
			const rendered = render(<Checkbox native />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			const inputElement = rendered.container.querySelector('input');

			expect(rendered.container).toBeInTheDocument();
			expect(checkboxElement).toBeInTheDocument();
			expect(checkboxElement).toBe(inputElement);
			expect(checkboxElement.className.match(/disabled/)).toBeFalsy();
			expect(inputElement).not.toHaveAttribute('disabled');
		});

		it('is not checked by default', () => {
			const rendered = render(<Checkbox native />);
			const inputElement = rendered.container.querySelector('input');

			expect(inputElement).toBeInTheDocument();
			expect(inputElement.checked).toBe(false);
		});

		it('does not render an icon when checked', () => {
			const rendered = render(<Checkbox native checked />);
			const svgElement = rendered.container.querySelector('svg');
			const inputElement = rendered.container.querySelector('input');

			expect(svgElement).not.toBeInTheDocument();
			expect(inputElement).toBeInTheDocument();
			expect(inputElement.checked).toBe(true);
		});

		it('fires onClick prop when clicked', () => {
			const clickFn = jest.fn();

			const rendered = render(<Checkbox native onClick={clickFn} />);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');

			userEvent.click(checkboxElement);
			expect(clickFn).toHaveBeenCalled();
		});

		it('respects the disabled prop', () => {
			const clickFn = jest.fn();
			const rendered = render(<Checkbox native disabled />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');

			expect(checkboxElement.className.match(/disabled/)).toBeTruthy();
			expect(checkboxElement).toHaveAttribute('disabled');

			userEvent.click(checkboxElement);
			expect(clickFn).not.toHaveBeenCalled();
		});

		it('renders with additional style using prop', () => {
			const style = {
				padding: '20px',
			};

			const rendered = render(<Checkbox native style={style} />);
			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			const styles = getComputedStyle(checkboxElement);

			expect(styles.padding).toBe(style.padding);
		});

		it('is themeable with ThemeProvider', () => {
			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Checkbox native />
				</ThemeProvider>
			);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			expect(checkboxElement).toHaveClass(globalTheme.components.checkbox.className);
		});

		it('is themeable with theme prop', () => {
			const rendered = render(<Checkbox native theme={propTheme} />);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			expect(checkboxElement).toHaveClass(propTheme.components.checkbox.className);
		});

		it('is themeable and theme prop overrides ThemeProvider', () => {
			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Checkbox native theme={propTheme} />
				</ThemeProvider>
			);

			const checkboxElement = rendered.container.querySelector('.ss__checkbox');
			expect(checkboxElement).toHaveClass(propTheme.components.checkbox.className);
			expect(checkboxElement).not.toHaveClass(globalTheme.components.checkbox.className);
		});
	});
});
