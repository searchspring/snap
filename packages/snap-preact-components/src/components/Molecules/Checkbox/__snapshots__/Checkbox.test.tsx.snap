// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component native can handle checked state internally 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native does not render an icon when checked 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native fires onClick prop when clicked 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native is not checked by default 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native is themeable and theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox props-theme-class emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox global-theme-class emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native is themeable with theme prop 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox props-theme-class emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native renders 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native renders with additional style using prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 20px;
}

<input
    class="ss__checkbox emotion-0"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native respects the disabled prop 1`] = `
<DocumentFragment>
  <input
    class="ss__checkbox ss__checkbox--disabled emotion-0"
    disabled=""
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component native respects the disabled prop 2`] = `
<DocumentFragment>
  <input
    class="ss__checkbox ss__checkbox--disabled emotion-0"
    disabled=""
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Checkbox Component styled Can enable/disable useAlly with disableA11y prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled Can enable/disable useAlly with disableA11y prop 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled can disableStyles 1`] = `
<DocumentFragment>
  <span
    aria-label=" unchecked checkbox"
    class="ss__checkbox"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled can handle checked state internally 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled can use iconColor prop instead of color 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  fill: blue;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--close-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M56 5.638l-22.362 22.362 22.362 22.362-5.638 5.638-22.362-22.362-22.362 22.362-5.638-5.638 22.362-22.362-22.362-22.362 5.638-5.638 22.362 22.362 22.362-22.362z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled fires onClick prop when clicked 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

<span
    aria-label=" unchecked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled fires onClick prop when clicked 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

<span
    aria-label=" unchecked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled is not checked by default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

<span
    aria-label=" unchecked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled is themeable and theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  fill: red;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox props-theme-class emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  fill: blue;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox global-theme-class emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled is themeable with theme prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  fill: red;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox props-theme-class emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled passes props to icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid blue;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  fill: blue;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--close-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M56 5.638l-22.362 22.362 22.362 22.362-5.638 5.638-22.362-22.362-22.362 22.362-5.638-5.638 22.362-22.362-22.362-22.362 5.638-5.638 22.362 22.362 22.362-22.362z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled renders 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

<span
    aria-label=" unchecked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled renders an icon when checked 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

.emotion-1 {
  width: calc(12px - 30%);
  height: calc(12px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled renders with additional style using prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
  padding: 20px;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

<span
    aria-label=" unchecked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled respects the disabled prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

<span
    aria-label="disabled unchecked checkbox"
    class="ss__checkbox ss__checkbox--disabled emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component styled respects the disabled prop 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 12px;
  width: 12px;
  border: 1px solid #333;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(12px - 30%);
  height: calc(12px - 30%);
}

<span
    aria-label="disabled unchecked checkbox"
    class="ss__checkbox ss__checkbox--disabled emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <span
      class="ss__checkbox__empty"
    />
  </span>
</DocumentFragment>
`;

exports[`Checkbox Component uses bocachica theme 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 18px;
  width: 18px;
  border: 1px solid #6d7175;
  border-radius: 3px;
}

.emotion-0.ss__checkbox--disabled {
  opacity: 0.7;
}

.emotion-0 .ss__checkbox__empty {
  display: inline-block;
  width: calc(18px - 30%);
  height: calc(18px - 30%);
}

.emotion-1 {
  fill: #6d7175;
  width: calc(18px - 30%);
  height: calc(18px - 30%);
  position: relative;
}

<span
    aria-label=" checked checkbox"
    class="ss__checkbox emotion-0"
    role="checkbox"
    ssa11y="true"
    tabindex="0"
  >
    <svg
      class="ss__icon ss__icon--check-thin ss__checkbox__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.771 40.395l33.749-33.749 4.48 4.48-38.229 38.229-17.771-17.771 4.48-4.48z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
